modules/actions/pool/update-on-chain-data-v3.ts

```diff
@@ -0,0 +1,45 @@
+import { prisma } from "../../../prisma/prisma-client";
+import { networkContext } from "../../network/network-context.service";
+import { SUPPORTED_POOL_TYPES } from "../../pool/lib/pool-on-chain-data.service";
+import { Chain } from "@prisma/client";
+import { fetchTokenPairDataV3Router } from "../../pool/lib/pool-on-chain-tokenpair-data";
+import { prismaBulkExecuteOperations } from "../../../prisma/prisma-util";
+
+
+//TODO: Move this job to pool-on-chain-data-service with the necessary adaptations
+export async function updateOnChainDataV3(chain: Chain){
```

Comment:
Two hints to make it better aligned with our current design:
1. Action should receive the configuration as attributes, so there are no networkContext, or config imports. It helps us with running things in separation and testing.
2. Would be great if you could change the name to something more telling about what the function is responsible for. Update on chain data won't work, too vague. Maybe sync-pool-token-balances, or something like that?